csharp/src/Ice/TcpEndpoint.cs

```diff
@@ -27,13 +28,20 @@ public override bool Equals(Endpoint? other)
             {
                 return true;
             }
+
             if (other is TcpEndpoint tcpEndpoint)
             {
+                if (HasCompressionFlag != tcpEndpoint.HasCompressionFlag)
+                {
+                    return false;
+                }
+
                 if (Timeout != tcpEndpoint.Timeout)
                 {
                     return false;
                 }
-                if (HasCompressionFlag != tcpEndpoint.HasCompressionFlag)
+
+                if (Transport != tcpEndpoint.Transport)
```

Comment:
We already compare the Transport in the base class.

Reply:
Fixed